pkg/credentials/filter/ring_buffer.go

```diff
@@ -0,0 +1,88 @@
+package filter
+
+import (
+	"math"
+)
+
+// RingBuffer implements a simple ring buffer with a static maximal length.
+type RingBuffer struct {
+	start         int
+	end           int
+	currentLength int
```

Comment:
It is the currentLength represents the actually contained number of bytes. The buffer stays the same size always but the number of bytes that are buffered in it will vary.

Reply:
Actually you do not initialize the buffer with a "length" but rather with a capacity. I renamed the parameter of the constructor function to reflect that.

Reply:
Actually I got rid of the end index now. It was only used in Add and the rest of the code used the length field. So now we only have the currentLength remaining. That means there should be no duplicate state here anymore.